designs/extensible-cli-and-scaffolding-plugins.md

```diff
@@ -21,9 +21,12 @@ Each plugin would minimally be required to implement the `Plugin` interface.
 
 ```go
 type Plugin interface {
-    // Version is the project version that this plugin implements.
-    // For example, Kubebuilder's Go v2 plugin implementation would return "2"
-    Version() string
+    // Version returns the project version that this plugin implements.
+    // For example, Kubebuilder's Go v2 plugin implementation would return 2.
+    Version() uint
```

Comment:
I think you should keep this as `Version() string` as that will allow more flexible versioning formats as `2.1.0` or `2.3-alpha1` similar to api version names-